### ISSUE TITLE
Support elasticsearch ssl verify configurable

### ISSUE THREAD

**Issue by TommyLike** [dadoonet/fscrawler#969]
**Is your feature request related to a problem? Please describe.**

There would be a case when the self generated certificate points to the internal IP address while it's beening visited via public IP, therefore the ssl verification would fail since the IP address does not match. 
![image](https://user-images.githubusercontent.com/5671225/85696563-a0e35e00-b70b-11ea-9e12-ded497aca502.png)

Requesting a feature that make this verification process configurable. for instance: here below is the configure option for fluentd on es ssl verifiation:
https://github.com/fluent/fluentd-kubernetes-daemonset/blob/master/fluentd-daemonset-elasticsearch-rbac.yaml#L75

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Comment by TommyLike** [comment]
**Update**:
In order to ignore the ssl hostname verification I updated the ESClient code as below:
```
private static RestClientBuilder buildRestClient(Elasticsearch settings) {
        List<HttpHost> hosts = new ArrayList<>(settings.getNodes().size());
        settings.getNodes().forEach(node -> hosts.add(HttpHost.create(node.decodedUrl())));

        RestClientBuilder builder = RestClient.builder(hosts.toArray(new HttpHost[hosts.size()]));

        if (settings.getPathPrefix() != null) {
            builder.setPathPrefix(settings.getPathPrefix());
        }

        if (settings.getUsername() != null) {
            CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
            credentialsProvider.setCredentials(AuthScope.ANY, new UsernamePasswordCredentials(settings.getUsername(), settings.getPassword()));
            builder.setHttpClientConfigCallback(httpClientBuilder ->
                    httpClientBuilder.setDefaultCredentialsProvider(credentialsProvider)).setHttpClientConfigCallback(
                           # ingore hostname verification 
                            httpClientBuilder-> httpClientBuilder.setSSLHostnameVerifier((s, sslSession) -> true));
        }

        return builder;
    }
```
Now fscrawler would not fail on ssl , but the strange thing would be it will always fail on authentication even I double checked the username and password:
curl command would pass
```
curl -v --user name:password   -k https://hostname:9200
```
while my fscrawler would fail:
```
elasticsearch:
  nodes:
   - url: "https://hostname:9200"
  index: "docs_test"
  bulk_size: 1000
  flush_interval: "5s"
  byte_size: "10mb"
  username: "admin"
  password: "password
```
![image](https://user-images.githubusercontent.com/5671225/85737771-3db8f200-b732-11ea-93ec-b5069d1030bd.png)

**Comment by TommyLike** [comment]
**UPDATE**:
oops, this is my first time to write JAVA code, and I think it's my mistake, the code above is incorrect, and I changed it into below and it works:
```
        if (settings.getUsername() != null) {
            CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
            credentialsProvider.setCredentials(AuthScope.ANY, new UsernamePasswordCredentials(settings.getUsername(), settings.getPassword()));
            builder.setHttpClientConfigCallback(httpClientBuilder -> {
                httpClientBuilder.setDefaultCredentialsProvider(credentialsProvider);
                httpClientBuilder.setSSLHostnameVerifier((s, sslSession) -> true);
                return httpClientBuilder;
            });
        }
```

**Comment by dadoonet** [comment]
Would you like to open a PR?

**Comment by TommyLike** [comment]
@dadoonet I am not sure if this change is acceptable since it's only ignored the ssl hostname verification.

**Comment by dadoonet** [comment]
I'd like to add this option though as it is surely useful for the community. Don't worry though, I'll take care of it if I have spare time. Let's just keep this issue opened.

**Comment by TommyLike** [comment]
@dadoonet I removed the ssl verification on my branched code here, still need to make it configurable:

https://github.com/opensourceways/fscrawler/blob/feature/support-ssl-verification-configurable/elasticsearch-client/elasticsearch-client-v7/src/main/java/fr/pilato/elasticsearch/crawler/fs/client/v7/ElasticsearchClientV7.java#L399-L414

**Comment by dadoonet** [comment]
Closed by #1036
